luigi/contrib/s3.py

```diff
@@ -163,236 +181,157 @@ def remove(self, path, recursive=True):
             return False
 
         (bucket, key) = self._path_to_bucket_and_key(path)
-
+        s3_bucket = self.s3.Bucket(bucket)
         # root
         if self._is_root(key):
-            raise InvalidDeleteException('Cannot delete root of bucket at path %s' % path)
-
-        # grab and validate the bucket
-        s3_bucket = self.s3.get_bucket(bucket, validate=True)
+            raise InvalidDeleteException(
+                'Cannot delete root of bucket at path %s' % path)
 
         # file
-        s3_key = s3_bucket.get_key(key)
-        if s3_key:
-            s3_bucket.delete_key(s3_key)
+        if self._exists(bucket, key):
+            self.s3.meta.client.delete_object(Bucket=bucket, Key=key)
             logger.debug('Deleting %s from bucket %s', key, bucket)
             return True
 
         if self.isdir(path) and not recursive:
-            raise InvalidDeleteException('Path %s is a directory. Must use recursive delete' % path)
+            raise InvalidDeleteException(
+                'Path %s is a directory. Must use recursive delete' % path)
 
-        delete_key_list = [
-            k for k in s3_bucket.list(self._add_path_delimiter(key))]
+        delete_key_list = []
+        for obj in s3_bucket.objects.filter(Prefix=self._add_path_delimiter(key)):
+            delete_key_list.append({'Key': obj.key})
 
         # delete the directory marker file if it exists
-        s3_dir_with_suffix_key = s3_bucket.get_key(key + S3_DIRECTORY_MARKER_SUFFIX_0)
-        if s3_dir_with_suffix_key:
-            delete_key_list.append(s3_dir_with_suffix_key)
+        if self._exists(bucket, key + S3_DIRECTORY_MARKER_SUFFIX_0):
+            delete_key_list.append({'Key': key + S3_DIRECTORY_MARKER_SUFFIX_0})
```

Comment:
Can we update the string combination here? (to not use `+`)

Reply:
Fixing now